### PR TITLE
Add videos as content type

### PR DESCRIPTION
**Pull request by katjuell**
### What should this PR do?
Resolves [DEVED-160](https://sourcegraph.atlassian.net/browse/DEVED-160) by adding a video content type to the site.

This includes:
- A `videos` index page
- A `posts` index page
- Reworking our existing tag template to a new template, `ArticleListTemplate`, that we can use whenever we are displaying a list of articles or videos.

This reorg does **not** move our dynamic `[slug]` template into the `videos` or `posts` directories for the following reasons:
- In the absence of running our own proxy with a `try_files` config, we are somewhat limited in the types of rewrites we are able to do, which we would need to redirect requests for top-level article `:slug`s. By keeping our dynamic `[slug]` template where it is, we can avoid that necessity, though, from a Next.js convention standpoint, it would perhaps make more sense to have videos under `/videos/[slug]` and posts under `/posts/[slug]`.
- Avoids any "soft" penalties on SEO or other unforeseen consequences.

### Why are we making this change?
- As we grow our content types, we want to make it easier for folks to navigate the repo. We would also like to be able to organize and display different content genres in creative ways.

### What are the acceptance criteria? 
- `/videos` displays a list of all of our videos.
- `/posts` displays a list of all of our posts.
- The **Load More** button for videos on the homepage takes you to the `videos` index.
- The **Videos** link on the main nav takes you to the `videos` index.
- The **Tutorials** link on the main nav takes you to the `posts` index.

### How should this PR be tested?
- Check out the branch, and ensure that the above criteria are satisfied.

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.

### CODE REVIEW

File: components/Header/index.tsx [hunk 1]
```diff
@@ -19,7 +19,9 @@ interface Props {
 
 const Header: FunctionComponent<Props> = props => (
     <>
-        <Column className='flex-large'>
+        <Column className={props.isHomepage
```
Comment (katjuell) [inline components/Header/index.tsx:22]: We need to toggle the `width` property in our header between the homepage and tag templates.

File: components/atoms/CollectionView/index.tsx [hunk 2]
```diff
@@ -25,11 +25,11 @@ const CollectionView: FunctionComponent<Props> = props => (
             <StyledCollectionTitle>{props.title}</StyledCollectionTitle>
         </StyledCollectionBody>
         <StyledCollectionList>
-            {props.members.map((post, index) => {
-                const isActive = props.activeSlug === post.slug
-                const titleText = `${index + 1}. ${post.frontMatter.title}`
+            {props.members.map((record, index) => {
```
Comment (katjuell) [inline components/atoms/CollectionView/index.tsx:28]: Semantic, but good to update.

File: components/atoms/ContentCardContainer/index.tsx [hunk 3]
```diff
@@ -34,7 +34,6 @@ const ContentCardContainer: FunctionComponent<Props> = props => (
                     imageAlt={record.frontMatter.imageAlt}
                     url={record.url}
                     key={record.url}
-                    contentType='post'
```
Comment (katjuell) [inline components/atoms/ContentCardContainer/index.tsx:37]: Don't actually use this is any meaningful way currently.

File: data/collections.yaml [hunk 4]
```diff
@@ -3,7 +3,13 @@ collections:
       slug: how-to-use-search-patterns-on-sourcegraph
       type: posts
       members:
-        - three-ways-to-search-code-with-sourcegraph
         - how-to-search-code-with-sourcegraph-using-literal-patterns
         - how-to-search-with-sourcegraph-using-regular-expression-patterns
         - how-to-search-with-sourcegraph-using-structural-patterns
+    - title: Understanding Git Basics
```
Comment (katjuell) [inline data/collections.yaml:9]: Open to discussion on this! Was using it mainly to test out behavior

Comment (ltagliaferri) [reply data/collections.yaml:9]: Yeah, I think this makes sense, I was thinking of adding more too.

**Review by ltagliaferri — APPROVED** [review]
LGTM 🚀

**Pull request merged**